### PR TITLE
Zeroth order wiki watcher.

### PR DESCRIPTION
**Pull request by sorensenjs**
Does not fetch the content of the change, but logs to
console changes as they occur.

### CODE REVIEW

File: antidox/wikiwatcher.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  Args:
+    change: a json object with the wikimedia change record.
+  """
+  print(
+      u'user:{user} namespace:{namespace} bot:{bot} comment:{comment} title:{title}'
+      .format(**change))
+
+
+def watcher(event_source, wiki_filter, namespaces_filter, callback):
+  """Watcher captures and filters evens from mediawiki.
```
Comment (jetpack) [inline antidox/wikiwatcher.py]: typo: evens -> events

File: antidox/wikiwatcher.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        continue
+      if change['namespace'] not in namespaces_filter:
+        continue
+      callback(change)
+
+
+if __name__ == '__main__':
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+      '--wiki_filter', default='enwiki', help='Wiki to scan, default enwiki.')
```
Comment (jetpack) [inline antidox/wikiwatcher.py]: For namespaces, you link to where they are defined, but what are the options for wiki_filter?

File: antidox/wikiwatcher.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      default='1,3',
+      help='Namespaces defined in http://phabricator.wikimedia.org/source/mediawiki/browse/master/includes/Defines.php separated by commas.'
+  )
+  parser.add_argument(
+      '--url',
+      default='https://stream.wikimedia.org/v2/stream/recentchange',
+      help='SSE client url')
+  args = parser.parse_args()
+
+  namespaces = set([int(ns) for ns in args.namespaces.split(',')])
```
Comment (jetpack) [inline antidox/wikiwatcher.py]: optional: you could use `parser.add_argument('--namespaces', nargs='+', type=int, ...)`.

https://docs.python.org/dev/library/argparse.html#nargs
( second answer, from Jakub M, here: https://stackoverflow.com/questions/15459997/passing-integer-lists-to-python )

File: antidox/wikiwatcher.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  """
+  for event in event_source:
+    if event.event == 'message' and event.data:
+      change = json.loads(event.data)
+      if change['bot']:
+        continue
+      if change['wiki'] != wiki_filter:
+        continue
+      if change['namespace'] not in namespaces_filter:
+        continue
```
Comment (jetpack) [inline antidox/wikiwatcher.py]: nit/optional: could combine these 3 checks w/ `or`

File: antidox/wikiwatcher.py (hunk 1 above)
Comment (sorensenjs) [reply antidox/wikiwatcher.py]: Done

File: antidox/wikiwatcher.py (hunk 2 above)
Comment (sorensenjs) [reply antidox/wikiwatcher.py]: That was done by inspection, I couldn't find a good reference.

File: antidox/wikiwatcher.py (hunk 3 above)
Comment (sorensenjs) [reply antidox/wikiwatcher.py]: Nice but can you use a default with a list? Seems no.
https://bugs.python.org/issue9625

File: antidox/wikiwatcher.py (hunk 4 above)
Comment (sorensenjs) [reply antidox/wikiwatcher.py]: Flipped and did and.

**Pull request merged**